flink-ml-parent/flink-ml-lib/src/main/java/org/apache/flink/ml/common/utils/TableUtil.java

```diff
@@ -0,0 +1,429 @@
+	}
+
+	/**
+	 * Find the types of the <code>targetCols</code>. If the targetCol not exist, return null.
+	 *
+	 * @param tableSchema TableSchema.
+	 * @param targetCols  the targetCols to find.
+	 * @return the corresponding types.
+	 */
+	public static TypeInformation[] findColTypes(TableSchema tableSchema, String[] targetCols) {
+		TypeInformation[] types = new TypeInformation[targetCols.length];
+		for (int i = 0; i < types.length; i++) {
+			types[i] = findColType(tableSchema, targetCols[i]);
+		}
+		return types;
+	}
+
+	/**
+	 * Find the type of the <code>targetCol</code>. If the targetCol not exist, return null.
+	 *
+	 * @param tableSchema TableSchema
+	 * @param targetCol   the targetCol to find.
+	 * @return the corresponding type.
+	 */
+	public static TypeInformation findColType(TableSchema tableSchema, String targetCol) {
+		int index = findColIndex(tableSchema.getFieldNames(), targetCol);
+
+		return index == -1 ? null : tableSchema.getFieldTypes()[index];
+	}
+
+	/**
+	 * Determine whether it is number type, number type includes double, long, byte, int, float and short.
+	 *
+	 * @param dataType the dataType to determine.
+	 * @return whether it is number type
+	 */
+	public static boolean isNumber(TypeInformation dataType) {
+		return Types.DOUBLE == dataType
+			|| Types.LONG == dataType
+			|| Types.BYTE == dataType
+			|| Types.INT == dataType
+			|| Types.FLOAT == dataType
+			|| Types.SHORT == dataType;
+	}
+
+	/**
+	 * Determine whether it is a string type.
+	 *
+	 * @param dataType the dataType to determine.
+	 * @return whether it is string type
+	 */
+	public static boolean isString(TypeInformation dataType) {
+		return Types.STRING == dataType;
+	}
+
+	/**
+	 * Determine whether it is a vector type.
+	 *
+	 * @param dataType the dataType to determine.
+	 * @return whether it is vector type
+	 */
+	public static boolean isVector(TypeInformation dataType) {
+		return VectorTypes.VECTOR.equals(dataType)
+			|| VectorTypes.DENSE_VECTOR.equals(dataType)
+			|| VectorTypes.SPARSE_VECTOR.equals(dataType)
+			;
+	}
+
+	/**
+	 * Check whether <code>selectedCols</code> exist or not, if not exist, throw exception.
+	 *
+	 * @param tableCols    a string array among which to find the target selectedCols.
+	 * @param selectedCols the selectedCols to assert.
+	 */
+	public static void assertSelectedColExist(String[] tableCols, String... selectedCols) {
+		if (null != selectedCols) {
+			for (String selectedCol : selectedCols) {
+				if (null != selectedCol) {
+					if (-1 == findColIndex(tableCols, selectedCol)) {
+						throw new IllegalArgumentException(" col is not exist " + selectedCol);
+					}
+				}
+			}
+		}
+	}
+
+	/**
+	 * Check whether colTypes of the <code>selectedCols</code> is numerical, if not, throw exception.
+	 *
+	 * @param tableSchema  TableSchema
+	 * @param selectedCols the selectedCols to assert.
+	 */
+	public static void assertNumericalCols(TableSchema tableSchema, String... selectedCols) {
```

Comment:
missing test for `null` type information

Reply:
We have added the test case.